### DISCUSSION TITLE
[ Feature Request ] Use arrow-down/up to rename next/previous file in list when renaming

### DISCUSSION THREAD

**Discussion by raimund-schluessler** [doublecmd/doublecmd#1147] [Suggestions]
Thanks a lot for creating/maintaining this nicely working file manager! I would like to propose one enhancement (of which I hope I didn't overlook a setting somewhere).

When renaming a file it would be very handy if one could select the next or previous file or folder by pressing the arrow down or up keys. The new name of the currently edited file would be saved and the next file would be open for renaming.

So the procedure would be:
- Start renaming a file/folder in a list with multiple files (e.g. by Shift+F6)
- Rename the file
- Press arrow-up/down
- The new name of the files gets saved and
- the next or previous file gets selected and is in renaming mode (without having to press Shift+F6 again)

Currently, the arrow-down/up buttons move the cursor in the input field, just like arrow left and right do. When renaming multiple files it would be faster to not have to press Shift+F6 for every file.

**Comment by ags1234** [comment]
This interesting and useful feature exists in Total Commander.

**Comment by goyalyashpal** [comment]
_moving over my modified input from other issue here:_

----
This feature of quick renaming in succession is also available in ms-explorer
- but using tab instead ... 
- i.e. the <kbd>tab</kbd> while renaming saves the rename, and
- cycles the highlight to other file in rename mode



 https://user-images.githubusercontent.com/19423063/216969024-c5b408be-824d-4a81-9e69-619764362894.mp4 



https://user-images.githubusercontent.com/19423063/217183195-b5081308-d637-469a-ad94-663a4fa56c59.mp4

**Comment by goyalyashpal** [comment]
_self hiding as resolved_

----
Above video shows _(edit: have replaced with a more simpler one):_
- Tab to cycle the highlighted file for rename, and
- F2 to cycle the cursor through parts of the filename itself i.e.
replace base-name, insert at end of base-name,  replace extension, replace full-name

Example of recent implementation of this F2 to cycle through rename selections is from '22.11 in [v1.71 of VS Code][f2-cycle-selections]

[f2-cycle-selections]: https://code.visualstudio.com/updates/v1_71#_explorer-rename-selection-improvements

----
**Update: 2023-02-12**
The F2 to cycle through is available in doublecmd 1.0.10 (239) too.

https://user-images.githubusercontent.com/19423063/218298044-de1ba8cd-64f7-45a9-b2fc-9a3c407a9f88.mp4

**Comment by goyalyashpal** [comment]
I would suggest against using up/down keys to cycle file selection as the arrow keys are used for cursor navigation in the rename text field itself:

- going to beginning or end (dc does it too)
- navigating up or down the multiline name lines 
since explorer shows long names in multiple lines when using "grid" or "Thumbnails View (`Shift+Ctrl+F1`)" ('dont know if dc can do this too!?)

**Comment by advice2020** [comment]
I will add my 2 cents about this also.
I agree this seems like a perfect feature to add to an orthodox style file manager (keyboard based).
Tricky part about it I would guess is picking a hotkey for it because it is not quite a command as others are where you could change its hotkey?  Changing the hotkey for it would have to be in its own section in the Options I would imagine?
I don't think anyone has mentioned it but why not F2 itself?
F2 to activate the inline rename and then up / down arrows to go to the previous / next file.
Not to say that it has to be the same, but I thought this is how it was done in Windows File Explorer which is the first time I have seen this in a file manager.  Thought that that worked pretty nice.

Another thing I don't think that was said is that this feature also provides another helpful purpose.
When you rename a file in DC if the beginning letter changes it auto sorts that filename taking you away from where the original file was, having this proposed feature would allow user the ability to be able to move cursor to the next / previous file in the same area of a file list if other files need to also be renamed.  Hopefully this makes sense how I am explaining it.

Anyways, as mentioned I agree, this one could be a great addition to DC if possible to add.
Thank You

**Comment by goyalyashpal** [comment]
> why not F2 itself?

i had mentioned that under a hidden comment above, have unhid now.

> F2 to activate the inline rename and then up / down arrows to go to the previous / next file.

i have already provided above that up/down is already used in dc while renaming.


I dont think u read above comments at all while there are just 4 of those....

**Comment by advice2020** [comment]
Ahh put a lot of restraint in the past not to comment to the reply above.
Most of my requests do not get any responses so I would happy if someone agreed with a request instead of the opposite.
Anyways, a while ago I had to use a Windows FM for something and was able to use this feature, it really emphasized how useful it could be.  This feature would fit right in with DCs keyboard based nature.  Really helpful to be able to keep the cursor in the same location of your file list when renaming the first letter of names, causing them to auto refresh/re-sort.
I am still interested in this one, hopefully this will be considered to be added in the future.
Thank You

EDIT:
@alexx2000 
As mentioned, I try not to call you out specifically but....
Alex!!!!!!!!!! this is amazing!  Thank you so much for adding this for all of us users interested in this feature.
Really appreciate it.

**Comment by alexx2000** [comment]
I added multi-rename using up / down arrows, many users requested exactly this variant.

**Comment by oliwerAR** [comment]
@alexx2000 
Is that included in 1.1.19? How does is work exactly?

**Reply by alexx2000** [reply]
No, only DC 1.2 has this feature.

**Comment by antodc** [comment]
@alexx2000 Is there any way to exit the rename mode without saving the changes other than clicking somewhere else? Apparently, if I start modifying a file name and for some reason I change my mind the only way to get out of the rename mode without approving the changes is to use the mouse. What about using ESC to go back to the initial name?

**Comment by alexx2000** [comment]
ESC should work.

**Comment by antodc** [comment]
I was using the following version and it did not work, but I can confirm that it works with the latest snapshot. Thank you!!

Double Commander
Version: 1.2.0 alpha
Revision: 11594
Commit: 904e910e8
Build date: 2024/10/20
Lazarus: 3.6.0.0
FPC: 3.2.2
Platform: x86_64-Win64-win32/win64
OS version: Windows 10 22H2 x86_64
